Backend/Remora.Discord.API.Abstractions/API/Objects/Activities/IActivity.cs

```diff
@@ -51,6 +51,7 @@ public interface IActivity
         /// <summary>
         /// Gets the time when the activity was added to the user's session.
         /// </summary>
+        // This field is modeled as optional, due to various inconsistencies in documentation, as well as real-world-observed behavior, for this field.
```

Comment:
This can be clarified (and should be placed in a `<remarks>` tag).

I'd suggest the following:

```
This field is always present on activities received from Discord; it is optional because it should never be sent by a bot.
```

Reply:
That's not the extent of it, though. I think it's also worth clarifying that the documentation is inconsistent with not only itself, but with real-world behavior.

Reply:
That the documentation is inconsistent is a concern for the docs repo (and an issue should be opened there). Which real-world behaviour have you seen that's not consistent with the docs (i.e, that it's always present on received payloads, but may be omitted when sending)?

Reply:
The field appears to be included in gateway messages, even when connected as a bot. Maybe there's some scenario where it's omitted, but that scenario is not simply "for bots" as the documentation states.

Yes, fixing documentation is a concern for the docs repo, but interpreting documentation is a Remora concern, and including a remark about why a particular field was modeled the way it was, when that decision is not obvious, is not only reasonable, but desirable. If only so the next guy who comes along and notices the inconsistency has a clear record of why it exists.

Reply:
I think you're misunderstanding the docs. Anything that comes from discord *will* have the field set. It may be omitted when we, as devs, send an activity (because it hasn't been created yet - that's why we're sending it). Discord's docs aren't very clearly laid out in this case, but they are unambiguous.

Reply:
That said, how about this:

```
This field is always present on activities received from Discord; it is optional because it should never be sent by a bot. Discord's documentation is somewhat vague at first glance, but because of the above it's implemented as an optional field.
```

Reply:
Aaaaaaaaaaallllright, I see what you mean. `Bots are only able to send name, type, and optionally url.` That's really worded weirdly. Another case of "If a model is different depending on where it's being used, just make separate freakin' models."

I'll update the PR tomorrow.